matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/network/RequestSender.kt

```diff
@@ -236,9 +246,44 @@ internal class RequestSender @Inject constructor(
                 .newBuilder()
                 .add(CheckNumberType.JSON_ADAPTER_FACTORY)
                 .build()
-                .adapter<Map<String, HashMap<String, Any>>>(Map::class.java)
+                .adapter<Map<String, Map<String, Any>>>(Map::class.java)
         val jsonBody = adapter.fromJson(body)!!
 
+        if (eventType == EventType.ROOM_KEY_REQUEST) {
+            scope.launch {
+                Timber.v("Intercepting key request, try backup")
+                /**
+                 * It's a bit hacky, check how this can be better integrated with rust?
```

Comment:
@yostyle is this a hack from us or from element android?

Reply:
from element